### PR TITLE
test: #29007 follow ups

### PR DESCRIPTION
**Pull request by 0xB10C**
A few, small follow-ups to #29007. See commit messages for details.

### CODE REVIEW

**Comment by 0xB10C** [comment]
Since `addpeeraddress` can currently fail to add an address to the `tried` table but will report `{ success: true }`, it makes sense to get that fixed with #28998 first before this PR, especially the first commit, is ready for review. The other two commits are minor and can wait.

File: src/addrdb.cpp [hunk 1]
```diff
@@ -204,15 +204,15 @@ util::Result<std::unique_ptr<AddrMan>> LoadAddrman(const NetGroupManager& netgro
         LogPrintf("Loaded %i addresses from peers.dat  %dms\n", addrman->Size(), Ticks<std::chrono::milliseconds>(SteadyClock::now() - start));
     } catch (const DbNotFoundError&) {
         // Addrman can be in an inconsistent state after failure, reset it
-        addrman = std::make_unique<AddrMan>(netgroupman, /*deterministic=*/deterministic, /*consistency_check_ratio=*/check_addrman);
+        addrman = std::make_unique<AddrMan>(netgroupman, deterministic, /*consistency_check_ratio=*/check_addrman);
         LogPrintf("Creating peers.dat because the file was not found (%s)\n", fs::quoted(fs::PathToString(path_addr)));
```
[the remark this answers is not in the document]
Comment (0xB10C) [reply src/addrdb.cpp:208]: I'm not doing this for now as I think this is slightly out-of-scope of this PR even if I touch the file.

**Comment by 0xB10C** [comment]
With https://github.com/bitcoin/bitcoin/pull/28998 merged, this PR is now ready-for-review.

**Review by stratospher — COMMENTED** [review]
tested ACK 9a44a20.

you can also include the check in `test/functional/feature_asmap.py` since `addpeeraddress` + deterministic addrman is present there too.

**Review by mzumsande — COMMENTED** [review]
Code Review ACK 9a44a20fb790f3be5d5d5d8f5d0f48aac633b2a4

**Pull request merged**